### PR TITLE
PR: Fix infinite loop with issue shorthand within comments

### LINKED ISSUE #11376: Infinite loop when "gl-N" (with a single digit) appears in multiline docstring
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

When typing a reference to an issue in a multiline docstring,
right after typing for example `gl-1` Spyder freezes, displaying only `gl-` and using 100% CPU.
Happens also with `gh-N` and `bb-N`.

The cursor stops blinking, the whole GUI stops responding, and a forced quit is necessary.

### What steps reproduce the problem?

1. Create blank file
2. Type in
```
"""
gl-1
```
3. Spyder hangs and uses 100% CPU.

### What is the expected output? What do you see instead?

We would expect to have an underlined piece of docstring with a Ctrl-clickable link, as when referencing a GitHub/GitLab/BitBucket issue in a simple string or in a comment.

### Paste Traceback/Error Below (if applicable)
No traceback is available as the GUI completely stops responding.

## Versions

* Spyder version: 4.0.1
* Python 3.7.6 64-bit 
* Qt 5.12.5 
* PyQt5 5.12.3 
* Linux 5.3.0-26-generic 

### Dependencies

```
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.2.2 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.11.1 (OK)
jedi =0.14.1                 :  0.14.1 (OK)
keyring                      :  None (OK)
nbconvert >=4.0              :  5.6.1 (OK)
numpydoc >=0.6.0             :  0.9.2 (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.7 (OK)
pygments >=2.0               :  2.5.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.5 (OK)
xdg >=0.26                   :  0.26 (OK)
zmq >=17                     :  18.1.1 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
qtawesome >=0.5.7            :  0.6.1 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
rtree >=0.8.3                :  0.9.3 (OK)
sphinx >=0.6.6               :  2.3.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  None (NOK)
matplotlib >=2.0.0           :  3.1.2 (OK)
numpy >=1.7                  :  1.17.3 (OK)
pandas >=0.13.1              :  None (NOK)
scipy >=0.17.0               :  1.3.3 (OK)
sympy >=0.7.3                :  1.5.1 (OK)
```

### PR DESCRIPTION
**Pull request by goanpeca**
### Issue(s) Resolved

Fixes #11376

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by ccordoba12** [comment]
@goanpeca, please add a regression test for this one.

**Comment by goanpeca** [comment]
Ok, working on it!

**Comment by goanpeca** [comment]
Fixed @ccordoba12

File: spyder/plugins/editor/widgets/tests/test_goto.py [hunk 1]
```diff
@@ -140,3 +140,17 @@ def interact():
     code_editor.filename = None
     code_editor._last_hover_pattern_key = None
     code_editor._last_hover_pattern_text = None
+
+
+def test_pattern_highlight_regression(qtbot, editorbot):
+    """Fix regression on spyder-ide/spyder#11376. """
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/tests/test_goto.py]: There's a blank space at the end of this docstring,

Comment (ccordoba12) [reply spyder/plugins/editor/widgets/tests/test_goto.py]: Ok, let's forget about this one.

Comment (goanpeca) [reply spyder/plugins/editor/widgets/tests/test_goto.py]: I missed this comment 😆

Comment (ccordoba12) [reply spyder/plugins/editor/widgets/tests/test_goto.py]: No biggie.

Comment (ccordoba12) [reply spyder/plugins/editor/widgets/tests/test_goto.py]: No biggie.

**Pull request merged**